Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4123,9 +4124,135 @@ - (void)setShouldObserveDBModifications:(BOOL)shouldObserveDBModifications
             CGPoint newContentOffset = CGPointMake(0, MAX(0, newContentHeight - viewTopToContentBottom));
             [self.collectionView setContentOffset:newContentOffset animated:NO];
         }
```

Comment:
We now do two separate things to update our view when we resume observing database modifications.

* We expand the mapping range to include any items added while we were not observing.
* We try to update scroll to reflect new items added while we were not observing.